src/ImageSharp/ApplyProcessors.cs

```diff
@@ -0,0 +1,107 @@
+{
+    using System;
+
+    using ImageSharp.PixelFormats;
+
+    using ImageSharp.Processing;
+
+    /// <summary>
+    /// Extension methods for the <see cref="Image{TPixel}"/> type.
+    /// </summary>
+    public static partial class ImageExtensions
+    {
+        /// <summary>
+        /// Mutates the image by applying the operations to it.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to rotate, flip, or both.</param>
+        /// <param name="operations">The operations to perform on the source.</param>
+        public static void Mutate<TPixel>(this Image<TPixel> source, Action<IImageOperations<TPixel>> operations)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            Guard.NotNull(operations, nameof(operations));
+
+            // TODO: add parameter to Configuration to configure how this is created, create an IImageOperationsFactory that cna be used to switch this out with a fake for testing
+            IImageOperations<TPixel> operationsRunner = source.Configuration.ImageOperationsProvider.CreateMutator(source);
+            operations(operationsRunner);
+        }
+
+        /// <summary>
+        /// Mutates the image by applying the operations to it.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to rotate, flip, or both.</param>
+        /// <param name="operations">The operations to perform on the source.</param>
+        public static void Mutate<TPixel>(this Image<TPixel> source, params IImageProcessor<TPixel>[] operations)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            Guard.NotNull(operations, nameof(operations));
+
+            // TODO: add parameter to Configuration to configure how this is created, create an IImageOperationsFactory that cna be used to switch this out with a fake for testing
+            IImageOperations<TPixel> operationsRunner = source.Configuration.ImageOperationsProvider.CreateMutator(source);
+            operationsRunner.ApplyProcessors(operations);
+        }
+
+        /// <summary>
+        /// Clones the current image mutating the clone by applying the operations to it.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to rotate, flip, or both.</param>
+        /// <param name="operations">The operations to perform on the source.</param>
+        /// <returns>Anew Image which has teh data from the <paramref name="source"/> but with the <paramref name="operations"/> applied.</returns>
+        public static Image<TPixel> Clone<TPixel>(this Image<TPixel> source, Action<IImageOperations<TPixel>> operations)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            Guard.NotNull(operations, nameof(operations));
+            var generated = new Image<TPixel>(source);
```

Comment:
I'm not sure if it's possible without changing `IImageProcessor<TPixel>` design.

The processors doing allocation of a new image buffer should be able to take the source and destination image buffers from the outside. It would make `SwapPixelBuffers()` unnecessary:
https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Processing/Processors/Transforms/ResizeProcessor.cs#L97

@JimBobSquarePants thoughts?